### PR TITLE
Add service deployment instructions to stable diffusion template

### PR DESCRIPTION
**Pull request by akshay-anyscale**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by akshay-anyscale** [comment]
Testing done using premerge: 

Screenshot 2023-07-06 at 11 47 31 AM
Screenshot 2023-07-06 at 11 46 03 AM
Screenshot 2023-07-06 at 11 45 26 AM

**Comment by akshay-anyscale** [comment]
Screenshot 2023-07-06 at 11 51 01 AM

**Review by mjconnor — APPROVED** [review]

**Review by justinvyu — COMMENTED** [review]
I think this generally looks good. A few suggestions:

- We can just remove the table at the start of the jupyter notebook, so that we don't need to update both.
- Could we also include the `serve run app:entrypoint && python query` in the test to make sure that at least works. I think testing the service is harder (is it possible to launch a service in a job?).

**Comment by akshay-anyscale** [comment]
@justinvyu where can I update the tests? Since this requires GPU to run should it still be part of the testing pipeline?

**Comment by justinvyu** [comment]
> @justinvyu where can I update the tests? Since this requires GPU to run should it still be part of the testing pipeline?

We can just add it at the end of the existing release test (which is setup with GPUs):

 https://github.com/ray-project/ray/blob/c67ebb45a7816e02273d7273edad74c77a9c600f/release/release_tests.yaml#L1003

**Comment by akshay-anyscale** [comment]
Successful release tests- https://buildkite.com/ray-project/release-tests-pr/builds/45230#01894726-bc4a-4684-952e-641a6b3fbc79

File: release/release_tests.yaml [hunk 1]
```diff
@@ -977,7 +977,7 @@
 
   run:
     timeout: 600
-    script: jupyter nbconvert --to script --output _test start.ipynb && ipython _test.py
+    script: jupyter nbconvert --to script --output _test start.ipynb && ipython _test.py && python app.py
```
Comment (justinvyu) [inline release/release_tests.yaml:980]: I think this needs to be:

```suggestion
    script: jupyter nbconvert --to script --output _test start.ipynb && ipython _test.py && serve run app:entrypoint && python query.py
```

File: doc/source/templates/03_serving_stable_diffusion/app.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        image = self.pipe(prompt, height=img_size, width=img_size).images[0]
+
+        return image
+
+
+entrypoint = APIIngress.bind(StableDiffusionV2.bind())
+
+# The following block will be executed if the script is run by Python directly
+if __name__ == "__main__":
+    serve.run(entrypoint)
```
Comment (justinvyu) [inline doc/source/templates/03_serving_stable_diffusion/app.py:61]: Don't think this block is needed since we go through `serve run`.

```suggestion
```

**Review by justinvyu — CHANGES_REQUESTED** [review]
Looks like the release test failed - made a suggestion fix. After that, let's run the release test again, and should be good to merge once it passes.

**Review by justinvyu — APPROVED** [review]
Thanks, LGTM.

Here was the passing release test for reference: https://buildkite.com/ray-project/release-tests-pr/builds/45812#01896c56-213c-4e4f-be52-f611eb96a03c

**Pull request merged**